src/generic/listctrl.cpp

```diff
@@ -4059,7 +4081,14 @@ void wxListMainWindow::RefreshAll()
 void wxListMainWindow::UpdateCurrent()
 {
     if ( !HasCurrent() && !IsEmpty() )
+    {
+        // Initialise m_current to the first item without sending any
+        // wxEVT_LIST_ITEM_FOCUSED event (typicaly when the control gains focus)
+        // and this is to allow changing the focused item using the arrow keys.
+        // which is the behaviour found in the wxMSW port.
+        wxEventBlocker block(GetParent(), wxEVT_LIST_ITEM_FOCUSED);
```

Comment:
Everything is under our control here, so it looks like it would be better to split the existing `ChangeCurrent()` into `ChangeCurrentWithoutEvent()` and call the new function and then `SendNotify()` from `ChangeCurrent()` itself. Then we could just call `ChangeCurrentWithoutEvent()` here without bothering with blocking anything.

Reply:
Done.